### PR TITLE
#2423 run only failed nrt

### PR DESCRIPTION
**Pull request by bongrainmat**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated
- [ ] if this commit targets a release branch: the corresponding milestone was added in the ticket and in this PR

### CODE REVIEW

File: dynawo/sources/Models/Modelica/Dynawo/Electrical/Machines/OmegaRef/GeneratorPV.mo [hunk 1]
```diff
@@ -40,7 +40,7 @@ model GeneratorPV "Generator with power / frequency modulation and voltage / rea
   final parameter Types.ReactivePowerPu QMinPu = QMin / SystemBase.SnRef "Minimum reactive power in pu (base SnRef)";
   final parameter Types.ReactivePowerPu QMaxPu = QMax / SystemBase.SnRef "Maximum reactive power in pu (base SnRef)";
   final parameter Types.PerUnit LambdaPu = LambdaPuSNom * SNom / SystemBase.SnRef "Reactive power sensitivity of the voltage regulation in pu (base UNom, SnRef)";
-  final parameter Types.Time T = 1 "Time constant used to filter the reactive power reference";
+  final parameter Types.Time T "Time constant used to filter the reactive power reference";
```
Comment (rosiereflo) [inline dynawo/sources/Models/Modelica/Dynawo/Electrical/Machines/OmegaRef/GeneratorPV.mo]: Not related

File: nrt/nrt.py [hunk 2]
```diff
@@ -582,6 +608,12 @@ def main():
                                 keep_job = False
                                 break
 
+                    if keep_job and (len(directory_failed) > 0):
+                        if job_file in directory_failed:
+                            keep_job = True
```
Comment (rosiereflo) [inline nrt/nrt.py:613]: The previous value of keep_job should be kept (if for example we only want to run a sublist of the failed tests)

**Review by rosiereflo — CHANGES_REQUESTED** [review]
Thanks!

File: nrt/nrt.py [hunk 3]
```diff
@@ -202,6 +202,7 @@ def launchCases(self):
 
 
     def exportHTML(self, html_output):
+        failed = []
```
[the remark this answers is not in the document]
Comment (bongrainmat) [reply nrt/nrt.py]: taken into account

File: nrt/nrt.py (hunk 2 above)
Comment (bongrainmat) [reply nrt/nrt.py]: taken into account

File: nrt/nrt.py [hunk 4]
```diff
@@ -531,6 +534,24 @@ def main():
...
         log_message += " pattern filter"
 
+    directory_failed = []
+    if (options.failed == True):
+        for root, dirs, files in os.walk(output_dir):
+            for file in files:
+                if "info.txt" == file:
+                    test_case = nrtDiff.TestCase("dummy")
+                    test_case.get_case_info(os.path.join(root, file))
+                    if test_case.status_ == "KO":
```
Comment (rosiereflo) [inline nrt/nrt.py:544]: I would propose to also add `or test_case.cmp_status == "KO" `as reference comparison failure is also a common reason to rerun nrt

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: nrt/nrt.py [hunk 5]
```diff
@@ -531,6 +534,24 @@ def main():
...
         log_message += " pattern filter"
 
+    directory_failed = []
+    if (options.failed == True):
+        for root, dirs, files in os.walk(output_dir):
+            for file in files:
+                if "info.txt" == file:
+                    test_case = nrtDiff.TestCase("dummy")
+                    test_case.get_case_info(os.path.join(root, file))
+                    if test_case.status_ == "KO" or test_case.cmp_status == "KO":
```
Comment (rosiereflo) [inline nrt/nrt.py:544]: should be "test_case.cmp_status_", you can easily test by taking some nrt results (nrt/outputs/branch_name) and modifying some info.txt files

**Review by rosiereflo — CHANGES_REQUESTED** [review]

**Review by rosiereflo — APPROVED** [review]

File: nrt/nrt.py [hunk 6]
```diff
@@ -531,6 +534,24 @@ def main():
 
         log_message += " pattern filter"
 
+    directory_failed = []
+    if (options.failed == True):
+        for root, dirs, files in os.walk(output_dir):
+            for file in files:
+                if "info.txt" == file:
+                    test_case = nrtDiff.TestCase("dummy")
```
Comment (marcochiaramello) [inline nrt/nrt.py:542]: what "dummy" stays for?

**Review by marcochiaramello — CHANGES_REQUESTED** [review]

File: nrt/nrt.py (hunk 6 above)
Comment (rosiereflo) [reply nrt/nrt.py:542]: the parsing works only if a testcase name was defined, so we provide a dummy one

Comment (marcochiaramello) [reply nrt/nrt.py:542]: ok, thx

**Review by marcochiaramello — APPROVED** [review]

**Pull request merged**